lxd/api_project.go

```diff
@@ -599,16 +599,18 @@ func projectPatch(d *Daemon, r *http.Request) response.Response {
 		req.Description = project.Description
 	}
 
-	config, err := reqRaw.GetMap("config")
-	if err != nil {
-		req.Config = project.Config
-	} else {
-		for k, v := range project.Config {
-			_, ok := config[k]
-			if !ok {
-				config[k] = v
+	// Perform config patch
+	req.Config = util.CopyConfig(project.Config)
+	patches, err := reqRaw.GetMap("config")
+	if err == nil {
+		for k, v := range patches {
+			strVal, ok := v.(string)
+			if ok {
+				req.Config[k] = strVal
 			}
 		}
+	} else {
+		logger.Error("Failed retrieving config map from request", logger.Ctx{"err": err})
```

Comment:
Does this fire if no input config is supplied by the request? 

I'm wondering why the current version has:

```go
config, err := reqRaw.GetMap("config")
	if err != nil {
		req.Config = project.Config
```

As I don't think we should be logging an error if a valid request with no config comes in for the patch request, we should just not updating anything in the config part.

Reply:
Good point -- with the incoming changes I do see the error logged if no input config is supplied. I agree that it's not a good idea to log an error here since the input is technically valid. Will fix this.